test/builder/builders.go

```diff
@@ -355,7 +362,9 @@ func (n *NodeBuilder) Build() *corev1.Node {
 			Name:      n.name,
 			Namespace: n.namespace,
 		},
-		Spec:   corev1.NodeSpec{},
+		Spec: corev1.NodeSpec{
```

Comment:
Wonder if we should set this only if `providerID` is present. Like this -
https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/9663cb8e24146dcec338c95228efff61f291f5c9/test/builder/builders.go#L76-L78

Reply:
No, Selector is pointer, providerID is string. Those value are all present, just whether initialize.
If a pointer is not initialize, it is nil, and it will coredump when you write data to nil address.
If a string is not initialize, it is empty.  It's no difference.